### PR TITLE
Add front/back networks

### PR DESCRIPTION
**Pull request by jeffballard**
If you want to use this on a directly-connected-to-the-Internet server,
putting Redis and friends directly on the Internet will typically
result in miscreants causing trouble.

This commit puts the backend containers on a backend network,
and the frontend container on both.

### CODE REVIEW

**Comment by tboerger** [comment]
I don't get why you need the additional networks.

**Comment by jeffballard** [comment]
Let me try again:

Prior to this PR, redis and mariadb will be directly connected to the external network when launched through docker-compose.  There are two main issues:

1. redis and mariadb will be on the Internet unfirewalled!  Both services do not need to be directly connectable from the Internet. (a security problem)
2. redis and/or mariadb will collide with any other redis/mariadb containers running on that node.

This PR moves the backend containers to their own private network, such that the frontend container can talk to them, and anything external cannot directly talk to them.

**Comment by tboerger** [comment]
Sorry, where did you get this information? `docker-compose` is always creating a separate network for every app launched via `docker-compose`, the network is never directly exposed to the internet.

I would really like to know how you get to this conclusion. We are running similar docker-compose.yml's on a daily basis and our services are never exposed to the internet, even without any firewall configured on our own.
